wallet/tests/e2e/specs/e2e-spec.js

```diff
@@ -0,0 +1,252 @@
+    });
+
+    it(`third deposit of value ${depositValue}`, () => {
+      // cy.get('#Bridge_amountDetails_tokenAmount').clear().type(depositValue);
+      cy.get('button').contains('Transfer').click();
+      cy.get('button').contains('Create Transaction').click();
+      cy.get('#Bridge_modal_continueTransferButton').click();
+      cy.wait(30000);
+      cy.confirmMetamaskTransaction().then(confirmed => expect(confirmed).to.be.true);
+      cy.wait(50000);
+      cy.get('.btn-close').click();
+    });
+
+    it(`fourth deposit of value ${depositValue}`, () => {
+      // cy.get('#Bridge_amountDetails_tokenAmount').clear().type(depositValue);
+      cy.get('button').contains('Transfer').click();
+      cy.get('button').contains('Create Transaction').click();
+      cy.get('#Bridge_modal_continueTransferButton').click();
+      cy.wait(30000);
+      cy.confirmMetamaskTransaction().then(confirmed => expect(confirmed).to.be.true);
+      cy.wait(50000);
+      cy.get('.btn-close').click();
+      cy.contains('Nightfall Assets').click();
+      cy.wait(20000);
+    });
+
+    it(`check token balance equal to ${depositValue * 4}`, () => {
+      cy.get('#TokenItem_tokenBalanceMATIC').should($div => {
+        const totalBalance = Number($div.text());
+        expect(totalBalance).to.equal(depositValue * 4);
+        currentTokenBalance = totalBalance;
+      });
+    });
+  });
+
+  context('Withdraw', () => {
+    const withdrawValue = 4;
+
+    it(`withdraw token of value ${withdrawValue}`, () => {
+      cy.get('#TokenItem_tokenWithdrawMATIC').click();
+      // cy.get('#Bridge_amountDetails_tokenAmount').clear().type(withdrawValue);
+      cy.get('label').contains('Withdraw').click();
+      cy.get('#Bridge_amountDetails_tokenAmount').type(withdrawValue);
+      cy.get('button').contains('Transfer').click();
+      cy.get('button').contains('Create Transaction').click();
+      cy.get('#Bridge_modal_continueTransferButton').click();
+      cy.wait(30000);
+      cy.confirmMetamaskTransaction().then(confirmed => expect(confirmed).to.be.true);
+      cy.wait(50000);
+      cy.get('.btn-close').click();
+      cy.contains('Nightfall Assets').click();
+    });
+
+    it(`check token balance after withdraw`, () => {
+      cy.get('#TokenItem_tokenBalanceMATIC').should($div => {
+        const totalBalance = Number($div.text());
+        expect(totalBalance).to.equal(currentTokenBalance - withdrawValue);
+        currentTokenBalance = totalBalance;
+      });
+    });
+  });
+
+  context('Single Transfer', () => {
+    const transferValue = 4;
+
+    /*
+     * dummy pkd of user who does not exist
+     * Note: even though we are passing recipientPkd but in code
+     *   it gets override by sender's pdk for now
+     *   hence two check
+     *   check1: before Block Proposed event
+     *   check2: after Block Proposed event
+     */
+    const recipientPkd = '0x90ff185f7fa35ddae731ddad18a958d55d45bb973c16735018f6bc6f3798a7e1';
```

Comment:
Similar test for a user using `nightfall-client` via the `SDK` is done in `test/ping-pong/user-local/src/index.mjs`. 
The approach might be slightly different in wallet tests but functionally they are the similar (minus the withdraw).
Can we replicate similar assignment to variables without hardcoding them?
Please also refer to `test/e2e/tokens/erc20.test.mjs` whilst bearing in mind the upcoming changes to these tests under (issue)[https://github.com/EYBlockchain/nightfall_3/issues/513]